pkg/kubectl-argo-rollouts/cmd/create/create.go

```diff
@@ -0,0 +1,279 @@
+		if err != nil {
+			return nil, err
+		}
+		fmt.Fprintf(c.Out, "%s.%s/%s created\n", rollouts.RolloutSingular, rollouts.Group, obj.Name)
+		return obj, nil
+	case gvk.Group == rollouts.Group && gvk.Kind == rollouts.AnalysisTemplateKind:
+		var template v1alpha1.AnalysisTemplate
+		err = unmarshal(fileBytes, &template)
+		if err != nil {
+			return nil, err
+		}
+		obj, err := c.RolloutsClientset().ArgoprojV1alpha1().AnalysisTemplates(ns).Create(&template)
+		if err != nil {
+			return nil, err
+		}
+		fmt.Fprintf(c.Out, "%s.%s/%s created\n", rollouts.AnalysisTemplateSingular, rollouts.Group, obj.Name)
+		return obj, nil
+	case gvk.Group == rollouts.Group && gvk.Kind == rollouts.AnalysisRunKind:
+		var run v1alpha1.AnalysisRun
+		err = unmarshal(fileBytes, &run)
+		if err != nil {
+			return nil, err
+		}
+		obj, err := c.RolloutsClientset().ArgoprojV1alpha1().AnalysisRuns(ns).Create(&run)
+		if err != nil {
+			return nil, err
+		}
+		fmt.Fprintf(c.Out, "%s.%s/%s created\n", rollouts.AnalysisRunSingular, rollouts.Group, obj.Name)
+		return obj, nil
+	default:
+		return nil, fmt.Errorf("creates of %s/%s unsupported", gvk.Group, gvk.Kind)
+	}
+}
+
+// NewCmdCreateAnalysisRun returns a new instance of an `rollouts create analysisrun` command
+func NewCmdCreateAnalysisRun(o *options.ArgoRolloutsOptions) *cobra.Command {
+	createOptions := CreateAnalysisRunOptions{
+		ArgoRolloutsOptions: *o,
+	}
+	var cmd = &cobra.Command{
+		Use:   "analysisrun",
+		Short: "Create an AnalysisRun from a template",
+		Example: o.Example(`
+  # Create an AnalysisRun from a local template file
+  %[1]s create analysisrun --from-file my-analysis-template.yaml
+  # Create an AnalysisRun from a template in the cluster
+  %[1]s create analysisrun --from my-analysis-template
+`),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			froms := 0
+			if createOptions.From != "" {
+				froms++
+			}
+			if createOptions.FromFile != "" {
+				froms++
+			}
+			if froms != 1 {
+				return fmt.Errorf("one of --from or --from-file must be specified")
+			}
+			templateArgs, err := createOptions.ParseArgs()
+			if err != nil {
+				return err
+			}
+			ns := o.Namespace()
+			var template *v1alpha1.AnalysisTemplate
+			if createOptions.From != "" {
+				template, err = createOptions.RolloutsClientset().ArgoprojV1alpha1().AnalysisTemplates(ns).Get(createOptions.From, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+			} else {
+				fileBytes, err := ioutil.ReadFile(createOptions.FromFile)
+				if err != nil {
+					return err
+				}
+				var tmpl v1alpha1.AnalysisTemplate
+				err = unmarshal(fileBytes, &tmpl)
+				if err != nil {
+					return err
+				}
+				template = &tmpl
+			}
+			var name, generateName string
+			if createOptions.Name != "" {
+				name = createOptions.Name
+			} else if createOptions.GenerateName != "" {
+				generateName = createOptions.GenerateName
+			} else {
+				generateName = template.Name + "-"
+			}
```

Comment:
Nix: I'm wondering if it would make the function easier to read if we make these lines into two helper functions:

```
template := getAnalysisTemplate(createOptions)
name, generateName := getAnalysisRunName(createOptions)
```

Reply:
I have no strong opinions either way so feel free to dismiss this comment

Reply:
sure.